scipy/optimize/optimize.py

```diff
@@ -1465,7 +1465,8 @@ def _minimize_newtoncg(fun, x0, args=(), jac=None, hess=None, hessp=None,
             A = fhess(*(xk,) + args)
             hcalls = hcalls + 1
 
-        while numpy.add.reduce(numpy.abs(ri)) > termcond:
+        k2 = 0
+        while (numpy.add.reduce(numpy.abs(ri)) > termcond) and (k2 < maxiter):
```

Comment:
As far as I remember the classical CG method is applicable if and only if the matrix is positive defined. If it does converge, then in n iterations: https://en.wikipedia.org/wiki/Conjugate_gradient_method

So the classical Newton-CG is applicable for convex problems only, when the Hessian is positive definite.

My suggestion is to change ``k2 < len(x0)``.